oracle/service.go

```diff
@@ -393,38 +388,43 @@ func (s *oracleSvc) commitSetPrices(dataC <-chan *PriceData) {
 			priceBatch = append(priceBatch, msg)
 		}
 
-		msgs := s.composeMsgs(priceBatch)
-		if len(msgs) == 0 {
-			batchLog.Debugf("pipeline composed no messages, so do nothing")
-			return
-		}
-
 		ts := time.Now()
-		txResp, err := s.cosmosClient.SyncBroadcastMsg(msgs...)
-		if err != nil {
-			metrics.ReportFuncError(s.svcTags)
-			batchLog.WithError(err).Errorln("failed to SyncBroadcastMsg")
-			return
-		}
+		for _, cosmoClient := range s.cosmosClients {
+			msgs := composeMsgs(cosmoClient, priceBatch)
+			if len(msgs) == 0 {
+				batchLog.Debugf("pipeline composed no messages, so do nothing")
+				return
+			}
 
-		if txResp.TxResponse != nil {
-			if txResp.TxResponse.Code != 0 {
+			txResp, err := cosmoClient.SyncBroadcastMsg(msgs...)
```

Comment:
So, we are sending prices to the chain for each cosmos client that we have? Aren't we sending a lot of duplicated info to the chain?

What if we just range over cosmos clients and, if one fails to send, we try with the next one. When one sends the prices with success, that's it, we can break the range. 

Or isn't this what's intended?

Reply:
I'm confused, It looks like Zé is right, we range over the list of cosmos available clients, compose the message using the same batch and then broadcast with each client.
We should use one client and then retry with the next one if fails